ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json

```diff
@@ -595,6 +595,56 @@
         }
       }
     },
+    {
+      "name": "Should repartition RHS if the key format comes from Schema Registry",
+      "statements": [
+        "CREATE TABLE left_table (l_id BIGINT PRIMARY KEY, name VARCHAR, foreign_key STRUCT<val BIGINT>) WITH (kafka_topic='left_topic', format='AVRO', partitions=128);",
+        "CREATE TABLE right_table (r_id STRUCT<val BIGINT> PRIMARY KEY, f1 INTEGER) WITH (kafka_topic='right_topic', format='AVRO', partitions=128);",
```

Comment:
I am surprise that this actually works? I was expecting `ON foreign_key = r_id` to fail as `STRUCT` should not be supported?